doc/api/module.md

```diff
@@ -111,6 +111,15 @@ changes:
 Register a module that exports [hooks][] that customize Node.js module
 resolution and loading behavior. See [Customization hooks][].
 
+> **Warning** If setting up a `MessageChannel` to communicate with a loader
+> within the Module Worker, beware that this can lead to a deadlock. For
+> example, you have 2 modules, A and B. "A" is registered first and sets up a
+> message channel,which it uses in its `resolve` hook. "B" uses `register` to
+> register its own loader. Resolving "B"'s own loader will go through "A"'s
```

Comment:
```suggestion
> register its own hooks. Resolving "B"'s own hooks will go through "A"'s
```

What does 'resolving B's own loader' mean?

Reply:
The import specifier supplied to `register()`. That import specifier goes through module resolution, and because loaders affect other loaders, loader A is involved, which creates the potential for a deadlock.